### PR TITLE
@W-10217427 add missing beforeLoad validation and remove redundant load()

### PR DESCRIPTION
**Pull request by lizaiv77**
- fix validation for beforeLoad: before the fix inside predicate any element could have been used
- remove redundant generated "load()" when generating element getter
- refactoring of tests for custom element getter

### CODE REVIEW

File: utam-compiler/src/main/java/utam/compiler/representation/CustomElementMethod.java [hunk 1]
```diff
@@ -102,15 +101,8 @@ public Single(
...
-          String.format(
-              "%s %s = %s.%s",
-              returnType.getSimpleName(), TMP_VARIABLE, builderPrefix, builderSuffix);
-      codeLines.add(firstStatement);
-      if (!isNullable && !isExternalImplementation) {
-        codeLines.add(String.format("%s.load()", TMP_VARIABLE));
-      }
-      codeLines.add("return " + TMP_VARIABLE);
+      String statement = String.format("return %s.%s", builderPrefix, builderSuffix);
+      codeLines.add(statement);
```
Comment (lizaiv77) [inline utam-compiler/src/main/java/utam/compiler/representation/CustomElementMethod.java:105]: load() in generated code is redundant - builder is taking care of invoking it

File: utam-core/src/main/java/utam/core/framework/base/BaseRootPageObject.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
   @Override
   public Object load() {
-    log("wait for a root element to be found");
+    log("load the object - wait for a root element to be found");
```
Comment (olivier-martin-sf) [inline utam-core/src/main/java/utam/core/framework/base/BaseRootPageObject.java:20]: Nitpick: the object feels very generic to me, what object are we talking about?

**Review by olivier-martin-sf — APPROVED** [review]
LGTM but I would ask someone else to review as I am not really familiar with the Java codebase

File: utam-core/src/main/java/utam/core/framework/base/BaseRootPageObject.java (hunk 2 above)
Comment (lizaiv77) [reply utam-core/src/main/java/utam/core/framework/base/BaseRootPageObject.java:20]: logger has a prefix that explicitly states what its applied to

Comment (olivier-martin-sf) [reply utam-core/src/main/java/utam/core/framework/base/BaseRootPageObject.java:20]: That's good then resolving this

**Pull request merged**